### PR TITLE
add support for alternate easyconfig parameters/templates/constants

### PR DESCRIPTION
**Pull request by jfgrimm**
As discussed in slack, this is an alternative approach to introducing new parameters and deprecating old ones in one go. Instead, it's a multi-stage process:
1. Introduce new names, as alternatives
2. Do the renaming
3. Deprecate
This would allow is to do it iteratively across the different parameters / etc. over a series of EB 5.x releases.

https://github.com/easybuilders/easybuild-framework/issues/4464

To play around with this, check out the branch and modify:
* `easybuild/framework/easyconfig/parser.py`:
```
 45 # alternate easyconfig parameters, and their non-deprecated equivalents                                                 
 46 ALTERNATE_PARAMETERS = {                                                                                                
 47     # <new_param>: <equivalent_param>,                                                                                  
 48 } 
 ```
 
* `easybuild/framework/easyconfig/templates.py`:
```
164 # alternate templates, and their equivalents                                                                            
165 ALTERNATE_TEMPLATES = {                                                                                                 
166     # <new>: <equivalent_template>,                                                                                     
167 }
```
```
174 # alternate template constants, and their equivalents                                                                   
175 ALTERNATE_TEMPLATE_CONSTANTS = {                                                                                        
176     # <new_template_constant>: <equivalent_template_constant>,                                                          
177 }
```

### CODE REVIEW

File: easybuild/framework/easyconfig/parser.py [hunk 1]
```diff
@@ -42,6 +42,11 @@
 from easybuild.tools.filetools import read_file, write_file
 
 
+# alternate easyconfig parameters, and their non-deprecated equivalents
+ALTERNATE_PARAMETERS = {
+    # <equivalent_param>: <new_param>,
```
Comment (jfgrimm) [inline easybuild/framework/easyconfig/parser.py:47]:
```suggestion
    # <new_param>: <equivalent_param> ,
```

File: easybuild/framework/easyconfig/templates.py [hunk 2]
```diff
@@ -161,6 +161,26 @@
     ('SHLIB_EXT', get_shared_lib_ext(), 'extension for shared libraries'),
 ]
 
+# alternate templates, and their equivalents
+ALTERNATE_TEMPLATES = {
+    # <equivalent_template>: <new_template>,
```
Comment (jfgrimm) [inline easybuild/framework/easyconfig/templates.py:166]:
```suggestion
    # <new_template>: <equivalent_template> ,
```

File: easybuild/framework/easyconfig/templates.py [hunk 3]
```diff
@@ -161,6 +161,26 @@
...
+}
+
+# deprecated templates, and their replacements
+DEPRECATED_TEMPLATES = {
+    # <old_template>: (<new_template>, <deprecation_version>),
+}
+
+# alternate template constants, and their equivalents
+ALTERNATE_TEMPLATE_CONSTANTS = {
+    # <equivalent_template_constant>: <new_template_constant>,
```
Comment (jfgrimm) [inline easybuild/framework/easyconfig/templates.py:176]:
```suggestion
    # <new_template_constant>: <equivalent_template_constant>,
```

**Comment by jfgrimm** [comment]
not really sure what I've done wrong, it works when you run the test in isolation...
```
FAIL: test_template_deprecation_and_alternate (test.framework.easyconfig.EasyConfigTest.test_template_deprecation_and_alternate)
Test deprecation of (and alternate) templates
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/runner/8eb0620a58092b2b260f5fc629e7607eac6df691/lib/python3.11/site-packages/test/framework/easyconfig.py", line 1498, in test_template_deprecation_and_alternate
    self.assertNotIn("%(" + tmpl + ")s", res)
AssertionError: '%(builddir)s' unexpectedly found in "cd %(start_dir)s && make %(namelower)s -Dbuild=%(builddir)s --with-cuda='%(cudaver)s' && echo %(alt_install_dir)s %(version_major_minor)s"
```

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**